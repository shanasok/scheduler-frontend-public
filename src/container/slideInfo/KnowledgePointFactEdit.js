import React from 'react';
import {bindActionCreators, compose} from "redux";
import {withRouter} from "react-router-dom";
import {connect} from "react-redux";
import {setCurrentLessonContent} from "../../redux/actions/lessonContentAction";
import TeacherSavePageContentIcon from "../teacher/TeacherSavePageContentIcon";


class KnowledgePointFactEdit extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            newVal: (this.props.content? this.props.content.fact : ""),
        };

        this.updateLocalState = this.updateLocalState.bind(this);
    }

    updateLocalState(event) {
        this.setState({newVal: event.target.value});
    }

    render() {

        const fact = (this.props.content)? this.props.content.fact : "";
        const elementIdForInput = "factEditBox"+this.props.contentId;

        return <div className="knowledgePoint">
            <TeacherSavePageContentIcon
                contentId={this.props.contentId}
                newValues={{
                    'fact':this.state.newVal
                }}
            />
            <input defaultValue={fact}
                   className="factEditBox"
                   id={elementIdForInput}
                   onChange={(e) => this.updateLocalState(e)}
            />

        </div>
    }
}

const mapStateToProps = state => ({
    currentLessonContent: state.currentLessonContent,
});

function mapDispatchToProps(dispatch) {
    return {
        actions: bindActionCreators({
            setCurrentLessonContent,
        }, dispatch)
    }
}

export default compose(
    withRouter,
    connect(mapStateToProps, mapDispatchToProps)
)(KnowledgePointFactEdit);
