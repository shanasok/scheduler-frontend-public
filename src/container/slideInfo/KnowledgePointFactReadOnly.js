import React from 'react';
import TeacherEditPageContentIcon from "../teacher/TeacherEditPageContentIcon";
import {bindActionCreators} from "redux";
import {connect} from "react-redux";

class KnowledgePointFactReadOnly extends React.Component {

    render() {
        const bulletPoint = '\u25CF';
        const fact = (this.props.content)? bulletPoint + ' ' + this.props.content.fact : "";
        const contentId = (this.props.content)? this.props.content.id : -1;
        const icons = (this.props.slideEditMode) ?
            <TeacherEditPageContentIcon pageContentId={contentId}/> : "";

        return <div className="knowledgePoint">
            {icons} <strong>{fact}</strong>
        </div>
    }
}

const mapStateToProps = state => ({
    slideEditMode: state.slideEditMode,
});

function mapDispatchToProps(dispatch) {
    return {
        actions: bindActionCreators({

        }, dispatch)
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(KnowledgePointFactReadOnly);