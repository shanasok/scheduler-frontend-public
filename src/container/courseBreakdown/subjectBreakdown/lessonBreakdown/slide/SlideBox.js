import React from 'react';
import {bindActionCreators} from "redux";
import KnowledgePointFactReadOnly from "../../../../slideInfo/KnowledgePointFactReadOnly";
import KnowledgePointDefinitionReadOnly from "../../../../slideInfo/KnowledgeDefinitionReadOnly";
import KnowledgePointFactEdit from "../../../../slideInfo/KnowledgePointFactEdit";
import KnowledgePointDefinitionEdit from "../../../../slideInfo/KnowledgeDefinitionEdit";
import {getPageContentFromLessonContent} from "../../../../../utils/findContent";
import {connect} from "react-redux";

import '../../../../../css/SlideStyles.css';

// export function getPageContent(lessonContent, pageNumber){
//     if (lessonContent && lessonContent.pages) {
//         const page = lessonContent.pages.filter(p => p.id === pageNumber);
//         if (page[0] && page[0].pageContent) {
//             return page[0].pageContent
//                 .sort((a, b) => (a.contentNumber > b.contentNumber) ? 1 : -1);
//         }
//     }
// }

export function getSlideKnowledgePoints(pageContent){
    if (pageContent && Array.isArray(pageContent)){
        return pageContent.map(individualContent => {
            return editOrReadOnlyKnowledgePoints(individualContent);
        });
    }
}

export function editOrReadOnlyKnowledgePoints(individualContent){
    if (individualContent.isEditable)
        return chooseKnowledgePointTypeEdit(individualContent.content)
    else
        return chooseKnowledgePointTypeReadOnly(individualContent.content);
}

export function chooseKnowledgePointTypeReadOnly(content){
    if (content && content.contentType === "FACT")
        return <KnowledgePointFactReadOnly key={'content'+content.id} content={content}/>
    else if (content && content.contentType === "DEFINITION")
        return <KnowledgePointDefinitionReadOnly key={'content'+content.id} content={content}/>
    else
        return <span>Unknown Data Type</span>;
}

export function chooseKnowledgePointTypeEdit(content){
    if (content && content.contentType === "FACT")
        return <KnowledgePointFactEdit
            key={'content'+content.id}
            contentId={content.id}
            content={content}
            fact={content.fact}
        />

    else if (content && content.contentType === "DEFINITION")
        return <KnowledgePointDefinitionEdit
            key={'content'+content.id}
            contentId={content.id}
            content={content}
        />
    else
        return <span>Unknown Data Type</span>;
}

class SlideBox extends React.Component {

    render() {
        const pageContent = getPageContentFromLessonContent(this.props.lessonContent, this.props.currentPage.pageNumber);
        const knowledgePoints = getSlideKnowledgePoints(pageContent);
        const slideMessage = (pageContent === undefined || pageContent.length === 0) ? "No content on this slide yet" : "";

        return <div className="slideBox">
            {slideMessage}
            {knowledgePoints}
        </div>
    }
}

const mapStateToProps = state => ({
    lessonContent: state.currentLessonContent,
    slideEditMode: state.slideEditMode,
});

function mapDispatchToProps(dispatch) {
    return {
        actions: bindActionCreators({

        }, dispatch)
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(SlideBox);