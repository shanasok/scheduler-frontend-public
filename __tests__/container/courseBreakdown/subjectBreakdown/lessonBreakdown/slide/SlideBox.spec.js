import React from "react";
import configureMockStore from "redux-mock-store";
import renderer from "react-test-renderer";
import {Provider as ReduxProvider} from "react-redux";
import {Router} from "react-router-dom";
import { createMemoryHistory } from "history";

import SlideBox, {
    chooseKnowledgePointTypeReadOnly,
    chooseKnowledgePointTypeEdit,
    getSlideKnowledgePoints,
    editOrReadOnlyKnowledgePoints
} from "../../../../../../src/container/courseBreakdown/subjectBreakdown/lessonBreakdown/slide/SlideBox";
import KnowledgePointDefinitionReadOnly from "../../../../../../src/container/slideInfo/KnowledgeDefinitionReadOnly";
import KnowledgePointDefinitionEdit from "../../../../../../src/container/slideInfo/KnowledgeDefinitionEdit";
import KnowledgePointFactReadOnly from "../../../../../../src/container/slideInfo/KnowledgePointFactReadOnly";
import KnowledgePointFactEdit from "../../../../../../src/container/slideInfo/KnowledgePointFactEdit";


it('renders correctly', () => {

    const pageContent1 = {
        contentNumber: 1,
        content: {
            contentType: "FACT",
            fact: "Test Fact",
            id: 1,
        }
    };
    const currentPage = {
        id: 1,
        pageNumber: 1,
        pageContent: [pageContent1],
    };

    const lesson = {
        id: 1,
        name: "test lesson name",
        lessonNumber: 1,
        pages: [currentPage],
    };

    const mockStore = configureMockStore()
    const mockState = {
        currentLessonContent: lesson,
        currentPage: currentPage,
    }
    let store = mockStore(mockState)

    const history = createMemoryHistory();
    history.location.state = {
        currentLessonContent: {
            id: 1,
            name: "Threading the Machine",
            lessonNumber: 1,
            pages:[currentPage],
        }
    };

    const tree = renderer
        .create(<ReduxProvider store={store}><Router history={history}>
            <SlideBox currentPage={currentPage}/>
        </Router></ReduxProvider>)
        .toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree.children[0]).toEqual("");

});

it('renders correctly with no pageContent', () => {

    const currentPage = {
        id: 1,
        pageNumber: 1,

    };

    const lesson = {
        id: 1,
        name: "test lesson name",
        lessonNumber: 1,
        pages: [currentPage],
    };

    const mockStore = configureMockStore()
    const mockState = {
        currentLesson: lesson,
        currentPage: currentPage,
    }
    let store = mockStore(mockState)

    const history = createMemoryHistory();
    history.location.state = {
        lesson: {
            id: 1,
            name: "Threading the Machine",
            lessonNumber: 1,
            pages:[],
        }
    };

    const tree = renderer
        .create(<ReduxProvider store={store}><Router history={history}>
            <SlideBox currentPage={currentPage}/>
        </Router></ReduxProvider>)
        .toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree.children[0]).toEqual('No content on this slide yet');
});


it('test chooseKnowledgePointTypeReadOnly where content is null', () => {
    const result = chooseKnowledgePointTypeReadOnly();
    expect(result).toEqual(<span>Unknown Data Type</span>);
});

it('test chooseKnowledgePointTypeReadOnly where content is definition', () => {
    const content = {
        contentType: "DEFINITION",
        slideEditable: false,
    }

    const result = chooseKnowledgePointTypeReadOnly(content);
    expect(result.type).toEqual(KnowledgePointDefinitionReadOnly);
});

it('test chooseKnowledgePointTypeReadOnly where content is fact', () => {
    const content = {
        contentType: "FACT",
        slideEditable: false,
    }

    const result = chooseKnowledgePointTypeReadOnly(content);
    expect(result.type).toEqual(KnowledgePointFactReadOnly);
});

it('test chooseKnowledgePointTypeEdit where content is null', () => {
    const result = chooseKnowledgePointTypeEdit();
    expect(result).toEqual(<span>Unknown Data Type</span>);
});

it('test chooseKnowledgePointTypeEdit where content is definition', () => {
    const content = {
        contentType: "DEFINITION",
        slideEditable: true,
    }

    const result = chooseKnowledgePointTypeEdit(content);
    expect(result.type).toEqual(KnowledgePointDefinitionEdit);
});

it('test chooseKnowledgePointTypeEdit where content is fact', () => {
    const content = {
        contentType: "FACT",
        slideEditable: true,
    }

    const result = chooseKnowledgePointTypeEdit(content);
    expect(result.type).toEqual(KnowledgePointFactEdit);
});

it('test editOrReadOnlyKnowledgePoints chooses edit fact correctly', () => {

    const pageContent = {
        isEditable: true,
        content: {
            contentType: "FACT",
            fact: "Test Fact1",
            id: 1
        },
    }

    const result = editOrReadOnlyKnowledgePoints(pageContent);
    expect(Array.isArray(result));
    expect(result.type).toEqual(KnowledgePointFactEdit);
});

it('test getSlideKnowledgePoints with dodgy content', () => {

    const pageContent = [{
        contentNumber: 1,
        content: {
            contentType: "UNKONWNTYPE",
            fact: "Test Unknown Type",
            id: 1,
        }
    }];

    const result = getSlideKnowledgePoints(pageContent);
    expect(Array.isArray(result));
    expect(result[0].type).toEqual('span');
});

it('test getSlideKnowledgePoints with FACT', () => {

    const pageContent = [{
        contentNumber: 1,
        content: {
            contentType: "FACT",
            fact: "Test Fact",
            id: 1,
        }
    }];

    const result = getSlideKnowledgePoints(pageContent);
    expect(Array.isArray(result));
    expect(result[0].type).toEqual(KnowledgePointFactReadOnly);
});

//Just commented this out coz method getPageContent is commented out and I can't remember why
// it('test getPageContent in correct order', () => {
//
//     const pageContent1 = {
//         contentNumber: 1,
//         content: {
//             contentType: "FACT",
//             fact: "Test Fact",
//             id: 1,
//         }
//     };
//     const pageContent2 = {
//         contentNumber: 2,
//         content: {
//             contentType: "FACT",
//             fact: "Test Fact2",
//             id: 2,
//         }
//     };
//
//     const currentPage = {
//         id: 1,
//         pageNumber: 1,
//         pageContent: [pageContent1, pageContent2],
//     };
//
//     const lesson = {
//         id: 1,
//         name: "test lesson name",
//         lessonNumber: 1,
//         pages: [currentPage],
//     };
//
//     const result = getPageContentFromLessonContent(lesson, 1);
//     expect(result).toEqual([pageContent1, pageContent2]);
//     expect(result).toMatchObject([pageContent1, pageContent2])
// });
//
// it('test getPageContent in correct order v2', () => {
//
//     const pageContent1 = {
//         contentNumber: 1,
//         content: {
//             contentType: "FACT",
//             fact: "Test Fact",
//             id: 1,
//         }
//     };
//     const pageContent2 = {
//         contentNumber: 2,
//         content: {
//             contentType: "FACT",
//             fact: "Test Fact2",
//             id: 2,
//         }
//     };
//
//     const currentPage = {
//         id: 1,
//         pageNumber: 1,
//         pageContent: [pageContent2, pageContent1],
//     };
//
//     const lesson = {
//         id: 1,
//         name: "test lesson name",
//         lessonNumber: 1,
//         pages: [currentPage],
//     };
//
//     const result = getPageContentFromLessonContent(lesson, 1);
//     expect(result).toEqual([pageContent1, pageContent2]);
//     expect(result).toMatchObject([pageContent1, pageContent2])
// });
//
//
// it('test getPageContent with no content', () => {
//
//     const lesson = {
//         id: 1,
//         name: "test lesson name",
//         lessonNumber: 1,
//         pages: [],
//     };
//
//     const result = getPageContentFromLessonContent(lesson, 1);
//     expect(result).toBeUndefined();
// });