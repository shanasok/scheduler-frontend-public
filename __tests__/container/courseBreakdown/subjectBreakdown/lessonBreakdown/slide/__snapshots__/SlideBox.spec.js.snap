// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="slideBox"
>
  
  <div
    className="knowledgePoint"
  >
    
     
    <strong>
      ‚óè Test Fact
    </strong>
  </div>
</div>
`;

exports[`renders correctly with no pageContent 1`] = `
<div
  className="slideBox"
>
  No content on this slide yet
</div>
`;
